Updates for moby 28.0 networking (#21612)
docker/docker.github.io: content/manuals/engine/network/packet-filtering-firewalls.md

addresses that exist within a host.

But, particularly with IPv6 you may prefer to avoid using NAT and instead
arrange for external routing to container addresses ("direct routing").

To access containers on a bridge network from outside the Docker host,
you must set up routing to the bridge network via an address on the Docker
host. This can be achieved using static routes, Border Gateway Protocol
(BGP), or any other means appropriate for your network.

Within a local layer-2 network, any remote host can set up a static route
to a container network via the docker host's address on the local network.

> So, those local remote hosts can access containers directly. Remote hosts
> outside the local network will only have direct access to containers if
> routers are configured to enable it. 

Reviewer: ```suggestion
This allows local remote hosts to access containers directly. For remote
hosts outside the local network, direct access to containers requires
router configuration to enable the necessary routing.
```

Author: Hmm, reading this again, "local remote hosts" are a bit weird!

How about I change it to ...

```
Within a local layer 2 network, remote hosts can set up static routes
to a container network using the Docker daemon host's address on the local
network. Those hosts can access containers directly. For remote
hosts outside the local network, direct access to containers requires
router configuration to enable the necessary routing.
```

Reviewer: Sounds good to me!

With mode `routed`, no NAT or masquerading rules are set up, but `iptables`
are still set up so that only published container ports are accessible.
Outgoing packets from the container will use the container's address,
not a host address.

In `nat` mode, when a port is published to a specific host address, that
port is only accessible via the host interface with that address. So,
for example, publishing a port to an address on the loopback interface
means remote hosts cannot access it.

However, using direct routing, published container ports are always
accessible from remote hosts, unless the Docker host's firewall has
additional restrictions. Hosts on the local layer-2 network can set up
direct routing without needing any additional network configuration.
Hosts outside the local network can only use direct routing to the
container if the network's routers are configured to enable it.

In `nat-unprotected` mode, unpublished container ports are also
accessible using direct routing, no port filtering rules are set up.
This mode is included for compatibility with legacy default behaviour.

The gateway mode also affects communication between containers that

> are connected to different docker networks on the same host.

Reviewer: ```suggestion
are connected to different Docker networks on the same host.
```

Author: Fixed.
